### PR TITLE
dev-python/rstcheck-3.3.1: Add rstcheck python package

### PR DESCRIPTION
**Pull request by samuelbernardo**
Signed-off-by: Samuel Bernardo <samuelbernardo.mail@gmail.com>

### CODE REVIEW

File: dev-python/rstcheck/rstcheck-3.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	$(python_gen_cond_dep 'dev-python/importlib_metadata[${PYTHON_USEDEP}]' pypy3 python3_{6,7})
```
Comment (DarthGandalf) [inline dev-python/rstcheck/rstcheck-3.3.1.ebuild:21]: Did you intend to use different python versions here than in PYTHON_COMPAT?

Comment (samuelbernardo) [reply dev-python/rstcheck/rstcheck-3.3.1.ebuild:21]: No, they should be the same.

File: dev-python/rstcheck/rstcheck-3.3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Checks syntax of reStructuredText and code blocks nested within it"
+HOMEPAGE="https://github.com/myint/rstcheck"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
```
Comment (DarthGandalf) [inline dev-python/rstcheck/rstcheck-3.3.1.ebuild:16]: Probably remove most of this: https://devmanual.gentoo.org/keywording/index.html

**Comment by samuelbernardo** [comment]
I did a mistake when pushing to this PR changes from other one. The last two commits shouldn't be here.
I will revert both.

**Comment by DarthGandalf** [comment]
You need to fix the issues pointed out in CI anyway.

Did you add the package to guru?

**Comment by samuelbernardo** [comment]
> You need to fix the issues pointed out in CI anyway.
> 
> Did you add the package to guru?

Yes, the package rstcheck was added first to guru.
I will look into this now. I think the only requirement is to update version.

**Comment by samuelbernardo** [comment]
> I will look into this now. I think the only requirement is to update version.
rstcheck-3.3.1 is still the latest version.
It as been already approved to guru master branch:
https://github.com/gentoo-mirror/guru/tree/master/dev-python/rstcheck

What I'm missing here?

**Comment by mjeveritt** [comment]
<veremitz> !note samuelbernardo you need to add a 'maintainer' section to metadata.xml with your name and often the proxy-maint project too

**Comment by scardracs** [comment]
@samuelbernardo ping: still interested on that PR?

**Comment by samuelbernardolip** [comment]
> @samuelbernardo ping: still interested on that PR?

Yes, just miss this. I'm going to correct the missing details.

**Review by scardracs — COMMENTED** [review]
In the meantime rstcheck reaches v6.1.1 https://pypi.org/project/rstcheck/

**Comment by samuelbernardo** [comment]
Closing this PR since it is completely outdated from master and after merge pkgcheck don't allow to use python_3.11.
Will create a new fresh PR for the new release.

**Pull request closed without merging**